Integration tests are broken on Go tip

It appears that something changed recently in Go tip that is causing the build to fail consistently:

https://travis-ci.org/golang/dep/jobs/299143402#L473

```
--- FAIL: TestIntegration/init/vndr/case1 (0.00s)
        --- FAIL: TestIntegration/init/vndr/case1/internal (0.31s)
        	testproj.go:112: go standard error:
        	testproj.go:113: initializing cache in $GOCACHE: mkdir /test-dep-home-does-not-exist: permission denied
```

--- a/cmd/dep/dep_test.go
+++ b/cmd/dep/dep_test.go
@@ -35,6 +35,9 @@ func TestMain(m *testing.M) {
 		os.Setenv("CCACHE_DIR", filepath.Join(home, ".ccache"))
 	}
 	os.Setenv("HOME", "/test-dep-home-does-not-exist")
+	if os.Getenv("GOCACHE") == "" {
+		os.Setenv("GOCACHE", "off") // because $HOME is gone
+	}
 
 	r := m.Run()
 

